### ISSUE TITLE
Having to expose the content property of relationships with computed properties. 

### ISSUE THREAD

**Issue by arenoir** [emberjs/data#3911]
All my computed properties that interact with relationships are broken in ember-2.1 - 2.2 with ember-data 2.0 - 2.2

For example:

```
//models/invoice.js
export default DS.Model({
  lines: DS.hasMany('invoice-line')
});
​
//models/invoice-line.js
export default DS.Model({
  amount: DS.attr('number')
});
​
//components/invoice.js
export default Ember.Component({
  total: computed('invoice.lines.@each.amount', function() {
    this.get('invoice.lines').reduce((value, item) => {
      return value += item.get('amount');
    }, 0);
  }),
});
```

That computed property `total` should work but does not.  The workaround is to expose the content property of the `invoice.lines` relationship. 

ie: 'invoice.lines.@each.amount' to 'invoice.lines.content.@each.amount' 

I however do not want to do that as this is "obviously?" a bug.

I am unsure if this is a bug with ember-data's RecordArray Promise implementation or ember's bug observing arrays.

**Comment by jcope2013** [comment]
related? https://github.com/emberjs/ember.js/issues/12475

**Comment by stefanpenner** [comment]
@jcope2013 I believe this is, at least partially, the source of at least some of the issues reported on emberjs/ember.js#12475

**Comment by arenoir** [comment]
Fixed in ember 2.2.3 by https://github.com/emberjs/ember.js/pull/12908
